### PR TITLE
Documentation for IAS -> Xsuaa token-xchange

### PR DESCRIPTION
**Pull request by liga-oz**

### CODE REVIEW

**Comment by nenaraab** [comment]
- avoid term "reuse service" (not official vocabulary)
- ###title -> ### title
- avoid "and XSUAA to support transition away from XSUAA" both will co-exist for a very long time 
- for readability i would prefer subsections instead of very long nested list... e.g. so instead of 1. (setup section):

```
### Setup
#### Establish Trust Between XSUAA and IAS
```

- sometimes you use 
```
1. 
  1.
```
and sometimes you use
```
1. 
  -
```
which results in another formatting of the list sub-items

File: docs/IAS-XSUAA-token-xchange.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+Call a secured endpoint with an IAS token
+1. Fetch the IAS token using the service key for the IAS service instance
+2. Call the secured endpoint with the IAS token like in the example below
+   ```shell script
+    curl --location --request GET 'https://myApp.cfapps.sap.hana.ondemand.com/securedEndpoint' \
+    --header 'Authorization: Bearer <FETCHED IAS TOKEN>'
+   ```
+You should receive an authorised response, if everything works fine.
+   
+##### Details
```
Comment (nenaraab) [inline docs/IAS-XSUAA-token-xchange.md:56]: -> ### Further Details

File: docs/IAS-XSUAA-token-xchange.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+### Under the hood
+![IAS -> XSUAA token xchange flow diagram](token-xchange.png)
+
+Token of incoming request is checked whether it is a Xsuaa token, if it is Xsuaa token then this token proceeds forward to the token validation step. In case it is valid token, successful response is sent back, otherwise an unauthorised response is sent. 
+In case token is not Xsuaa token, application checks if token exchange is enabled, if it is, then Xsuaa access token is requested from Xsuaa instance using the POST request with Header `X-zid`:`zone_uuid`, `grant_type` = `jwt-bearer` and `assertion` = IAS token from the request. Upon successful token exchange between IAS and XSUAA, token gets validated and depending on validation result authorised or unauthorised response is sent back.
+In situation when incoming request doesn't contain Xsuaa token and token exchange is disabled, unauthorised response is sent back. 
+### Setup
+#### Setup trust between IAS and Xsuaa 
+Detailed information can be found [here](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/161f8f0cfac64c4fa2d973bc5f08a894.html)
+1. create an XSUAA service instance with plan apiaccess in your subaccount
```
Comment (nenaraab) [inline docs/IAS-XSUAA-token-xchange.md:20]: Start uppercase for list item "Create...", "Fetch..."

File: docs/IAS-XSUAA-token-xchange.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+### Scenario
+The user is logged in to a multitenant application that has adopted IAS and zones.
+The application (or an internally used SCP service instance) calls a service in the context of the current user and zone, by sending the IAS ID token for the user. That means the IAS ID token consists of `user_uuid` and `zone_uuid` claim.
+The service instance uses XSUAA internally and thus needs to exchange the ID token from IAS to an access token from XSUAA. 
+In the XSUAA multitenancy concept, the zone is mapped to the single XSUAA tenant that belongs to the zone (=> XSUAA tenant ID = zone ID).
+
+### Under the hood
+![IAS -> XSUAA token xchange flow diagram](token-xchange.png)
+
+Token of incoming request is checked whether it is a Xsuaa token, if it is Xsuaa token then this token proceeds forward to the token validation step. In case it is valid token, successful response is sent back, otherwise an unauthorised response is sent.
```
Comment (nenaraab) [inline docs/IAS-XSUAA-token-xchange.md:14]: authorised, authorisation => use american notion of it: authorized, authorization

File: docs/IAS-XSUAA-token-xchange.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+In situation when incoming request doesn't contain Xsuaa token and token exchange is disabled, unauthorised response is sent back. 
+### Setup
+#### Setup trust between IAS and Xsuaa 
+Detailed information can be found [here](https://help.sap.com/viewer/65de2977205c403bbc107264b8eccf4b/Cloud/en-US/161f8f0cfac64c4fa2d973bc5f08a894.html)
+1. create an XSUAA service instance with plan apiaccess in your subaccount
+2. fetch access token from this service instance
+3. execute the trust setup:
+  ```shell script
+    curl --location --request POST 'https://api.authentication.sap.hana.ondemand.com/sap/rest/identity-providers' \
+    --header 'Authorization: bearer <FETCHED XSUAA ACCESS TOKEN>' \
```
Comment (nenaraab) [inline docs/IAS-XSUAA-token-xchange.md:25]: is curl still required, when you did as explained in help.sap.com? Maybe you can ask Patrick?

**Review by nenaraab — CHANGES_REQUESTED** [review]
That's all from my side... need to ask others for further feedback...

**Review by nenaraab — COMMENTED** [review]
👍 thanks - lets wait for the feedback from the others :-)

File: docs/IAS-XSUAA-token-xchange.md (hunk 4 above)
Comment (liga-oz) [reply docs/IAS-XSUAA-token-xchange.md:27]: confirmed by Patrick both do the same thing. Thanks for pointing out 💯

**Review by nenaraab — APPROVED** [review]
from my side, this looks very good. Thanks a lot for the efforts!

File: docs/IAS-XSUAA-token-xchange.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+   ```
+#### 4. Call secured endpoint with an IAS token
+Call the secured endpoint with the IAS token like in the example below
+   ```shell script
+    curl GET 'https://yourApp.cfapps.sap.hana.ondemand.com/yourSecuredEndpoint' \
+    --header 'Authorization: Bearer <FETCHED IAS TOKEN>'
+   ```
+You should receive an authorized response, if everything works fine.
+
+:interrobang: In case of 403 error, check if the user has required role assigned.
```
Comment (nenaraab) [inline docs/IAS-XSUAA-token-xchange.md:55]: Hi @liga-oz does the role have to be assigned to "sap.custom" or "sap.default" origin or both?

Comment (liga-oz) [reply docs/IAS-XSUAA-token-xchange.md]: Hi @nenaraab, role has to be assigned to IAS tenant that has established trust with the XSUAA i.e. the IAS instance that has application in the space where Xsuaa instance and application were created. In your instance it's sap.custom origin, but the origin naming can vary, based on service setup.

**Pull request merged**